File: backend/app/Models/Producto.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Producto extends Model
{
    use HasFactory;

    protected $fillable = [
        'nombre',
        'descripcion',
        'precio',
        'stock',
        'imagen',
        'categoria_id',
    ];

    protected $casts = [
        'precio' => 'decimal:2',
    ];

    /**
     * Obtener la categoría del producto
     */
    public function categoria()
    {
        return $this->belongsTo(Categoria::class);
    }

    /**
     * Obtener los detalles de pedidos que incluyen este producto
     */
    public function pedidoDetalles()
    {
        return $this->hasMany(PedidoDetalle::class);
    }
}

File: admin-app/CrepesCoffeeAdmin/Models/Producto.cs
using System;
using Newtonsoft.Json;

namespace CrepesCoffeeAdmin.Models
{
    public class Producto
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("nombre")]
        public string Nombre { get; set; } = string.Empty;

        [JsonProperty("descripcion")]
        public string? Descripcion { get; set; }

        [JsonProperty("precio")]
        public decimal Precio { get; set; }

        [JsonProperty("stock")]
        public int Stock { get; set; }

        [JsonProperty("imagen")]
        public string? Imagen { get; set; }

        [JsonProperty("categoria_id")]
        public int CategoriaId { get; set; }

        [JsonProperty("categoria")]
        public Categoria? Categoria { get; set; }

        [JsonProperty("created_at")]
        public DateTime CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime UpdatedAt { get; set; }

        public string PrecioFormateado => $"${Precio:F2}";
        public string StockStatus => Stock > 0 ? $"En Stock ({Stock})" : "Sin Stock";
    }
} 